src/Microsoft.VisualStudio.LanguageServices.Razor/TagHelperDescriptorJsonConverter.cs

```diff
@@ -0,0 +1,194 @@
+            if (descriptorKind != ITagHelperDescriptorBuilder.DescriptorKind)
+            {
+                throw new NotSupportedException(
+                    Resources.FormatTagHelperDescriptorJsonConverter_UnsupportedTagHelperDescriptorKind(
+                        typeof(TagHelperDescriptor).Name, descriptorKind));
+            }
+
+            var typeName = descriptor[nameof(TagHelperDescriptor.Name)].Value<string>();
+            var assemblyName = descriptor[nameof(TagHelperDescriptor.AssemblyName)].Value<string>();
+            var tagMatchingRules = descriptor[nameof(TagHelperDescriptor.TagMatchingRules)].Value<JArray>();
+            var boundAttributes = descriptor[nameof(TagHelperDescriptor.BoundAttributes)].Value<JArray>();
+            var childTags = descriptor[nameof(TagHelperDescriptor.AllowedChildTags)].Value<JArray>();
+            var documentation = descriptor[nameof(TagHelperDescriptor.Documentation)].Value<string>();
+            var tagOutputHint = descriptor[nameof(TagHelperDescriptor.TagOutputHint)].Value<string>();
+            var diagnostics = descriptor[nameof(TagHelperDescriptor.Diagnostics)].Value<JArray>();
+            var metadata = descriptor[nameof(TagHelperDescriptor.Metadata)].Value<JObject>();
+
+            var builder = ITagHelperDescriptorBuilder.Create(typeName, assemblyName);
+
+            builder
+                .Documentation(documentation)
+                .TagOutputHint(tagOutputHint);
+
+            foreach (var tagMatchingRule in tagMatchingRules)
+            {
+                var rule = tagMatchingRule.Value<JObject>();
+                builder.TagMatchingRule(b => ReadTagMatchingRule(b, rule, serializer));
+            }
+
+            foreach (var boundAttribute in boundAttributes)
+            {
+                var attribute = boundAttribute.Value<JObject>();
+                builder.BindAttribute(b => ReadBoundAttribute(b, attribute, serializer));
+            }
+
+            foreach (var childTag in childTags)
+            {
+                var tagValue = childTag.Value<string>();
+                builder.AllowChildTag(tagValue);
+            }
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var diagnosticReader = diagnostic.CreateReader();
+                var diagnosticObject = serializer.Deserialize<RazorDiagnostic>(diagnosticReader);
+                builder.AddDiagnostic(diagnosticObject);
+            }
+
+            var metadataReader = metadata.CreateReader();
+            var metadataValue = serializer.Deserialize<Dictionary<string, string>>(metadataReader);
+            foreach (var item in metadataValue)
+            {
+                builder.AddMetadata(item.Key, item.Value);
+            }
+
+            return builder.Build();
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            throw new NotImplementedException();
```

Comment:
Technically, the default behavior is being used for serialization. `public override bool CanWrite => false;` takes care of that. As `JsonConverter` is an abstract class, I can't call `base.WriteJson` on this and I can't not override `WriteJson` method.